internal/parse/parse.go

```diff
@@ -37,6 +37,56 @@ var (
 	errBadVolSrc    = errors.New("must set volume source")
 )
 
+// GetBuildOutput is responsible for parsing custom build output argument.
+// Takes `buildOutput` as string and returns path where output has to be generated.
+func GetBuildOutput(buildOutput string) (string, bool, bool, error) {
+	isDir := true
+	isStdout := false
+	typeSelected := false
+	pathSelected := false
+	path := ""
+	if len(buildOutput) == 1 && buildOutput == "-" {
+		// Feature parity with buildkit, output tar to stdout
+		// Read more here: https://docs.docker.com/engine/reference/commandline/build/#custom-build-outputs
+		return "", false, true, nil
+	}
+	if strings.Contains(buildOutput, ",") {
+		tokens := strings.SplitN(buildOutput, ",", 2)
+		if len(tokens) != 2 {
```

Comment:
Is this somehow a fundamental design constraint? A compatibility constraint? The format at seems to support an arbitrary number of options just fine.

If this is to support a `,` character inside a path, please explicitly document that in a comment here (and perhaps even in the man page). (But I’m not sure about that idea, it limits us from adding more option types in the future.)

Reply:
We are now erroring out for every unknown option

Reply:
That’s not quite what I meant.

With this, `type=tar,dest=/tmp/with,option=comma` parses the path as `/tmp/with,option=comma`, which is both undocumented and making it impossible for us to add any more options (after a `dest` option, or ever? undocumented behavior).

I’d suggest using SplitN(…3) at least, to forbid commas in values, or preferably just using `Split()`, removing this length check completely, and relying on the duplicate checks, and on after-the-loop checks to verify all data has been provided.

Reply:
fixed.